### ISSUE TITLE
`autodoc_default_flags` deprecated on a minor version

### ISSUE THREAD

**Issue by vrde** [sphinx-doc/sphinx#5421]
*Disclaimer: I assume Sphinx follows the [Semantic Versioning rules](https://semver.org/).*

Subject: Sphinx 1.8 is not backward compatible since it deprecates `autodoc_default_flags`.

### Problem
- Our CI fails to build our documentation, because of: https://github.com/sphinx-doc/sphinx/blob/efdc95ef8b041a90096d43144fad7086b1f93cc9/CHANGES#L149

#### Procedure to reproduce the problem
I'll skip this section, if you need the commands I can add them later.

#### Error logs / results
```
Warning, treated as error:
autodoc_default_flags is now deprecated. Please use autodoc_default_options instead.
```
- [Logs on Travis-CI](https://travis-ci.org/bigchaindb/bigchaindb/jobs/428062947#L518)

#### Expected results
If Sphinx follows the [Semantic Versioning rules](https://semver.org/), it shouldn't error out if `autodoc_default_flags` is used. Removing the option is a breaking change and should happen in the next major version of Sphinx (2.0?).

### Reproducible project / your project
- https://github.com/bigchaindb/bigchaindb/commit/4b54e702f8c2477a8bd56a8dffeda25b241a88a0

### Environment info
- OS: GNU/Linux (dunno what they use on Travis-CI, it fails on my Ubuntu 18.04 machine as well)
- Python version: 3.6
- Sphinx version: 1.8

**Comment by tk0miya** [comment]
Thank you for reporting.

We've not adopted to semantic versioning yet. But this warning has not obey to our deprecation rule.
It should be omitted during 1.8.x series.
http://www.sphinx-doc.org/en/master/devguide.html#deprecation-policy

I marked this as a bug and will fix this later.

**Comment by tk0miya** [comment]
Fixed by #5424.
Thank you for reporting!
